### PR TITLE
Disable failing torchscript tests

### LINKED ISSUE #624: Failures in `test_torchscript.py` using PyTorch main
CI is currently failing since it uses `main` on PyTorch. Recently a change was merged to PyTorch (https://github.com/pytorch/pytorch/pull/105072) that fixes a bug where importing `jit_metaprogramming_utils` caused the default dtype to be set to `torch.float64` instead of the default `torch.float32`. This is a welcome change since that global change interfered with other tests. After this, we should be able to safely run `pytest python_tests/` without hacky changes to enforce the default "default" dtype.

Unfortunately some of the tests in `test_torchscript.py` broke with this change; namely `test_category_rule`, but also some others (listed below):
- `test_category_rule`
- `test_alias_pass_fix`
- `test_conv2d_bias`
- `test_flatten`
- `test_index_select_shape_expression`
- `test_linear`
- `test_nested_view`
- `test_shape_expression`
- `test_squeeze`
- `test_squeeze_negative_dim`
- `test_squeeze_zero`
- `test_unsqueeze`
- `test_view`
- `test_view_copy_graph_guard`
- `test_view_copy_graph_guard_double_fusion`

Most of these are not fixed by setting the default dtype back to `torch.float64`. For example `test_conv2d_bias` fails at `self.assertGraphContains(graph, FUSION_GROUP, True)`. It seems like the graph changed significantly: on nvfuser a1395f75db with PyTorch `2.1.0a0+gite132f09` we have
```
python_tests/test_torchscript.py::TestCudaFuser::test_conv2d_bias graph(%x.1 : Tensor,
      %w.1 : Tensor,
      %bias.1 : Tensor):
  %3 : int[] = prim::Constant[value=[1, 1]]()
  %4 : int[] = prim::Constant[value=[0, 0]]()
  %5 : int = prim::Constant[value=1]()
  %6 : NoneType = prim::Constant()
  %o.1 : Tensor = aten::conv2d(%x.1, %w.1, %6, %3, %4, %3, %5) # /opt/pytorch/nvfuser/python_tests/test_torchscript.py:4117:16
  %8 : bool = prim::CudaFusionGuard[types=[Float(4, 2, 2, 2, strides=[8, 4, 2, 1], requires_grad=0, device=cuda:0), Float(2, strides=[1], requires_grad=0, device=cuda:0)]](%o.1, %bias.1)
  %9 : Tensor = prim::If(%8)
    block0():
      %10 : Tensor = prim::CudaFusionGroup_0[cache_id=0](%o.1, %bias.1)
      -> (%10)
    block1():
      %11 : Function = prim::Constant[name="fallback_function", fallback=1]()
      %12 : (Float(4, 2, 2, 2, strides=[8, 4, 2, 1], requires_grad=0, device=cuda:0)) = prim::CallFunction(%11, %o.1, %bias.1)
      %13 : Float(4, 2, 2, 2, strides=[8, 4, 2, 1], requires_grad=0, device=cuda:0) = prim::TupleUnpack(%12)
      -> (%13)
  return (%9)
with prim::CudaFusionGroup_0 = graph(%0 : Float(4, 2, 2, 2, strides=[8, 4, 2, 1], requires_grad=0, device=cuda:0),
      %1 : Float(2, strides=[1], requires_grad=0, device=cuda:0)):
  %2 : int = prim::Constant[value=-1]()
  %3 : Float(2, 1, strides=[1, 1], requires_grad=0, device=cuda:0) = aten::unsqueeze_copy(%1, %2)
  %4 : Float(2, 1, 1, strides=[1, 1, 1], requires_grad=0, device=cuda:0) = aten::unsqueeze_copy(%3, %2)
  %5 : Float(4, 2, 2, 2, strides=[8, 4, 2, 1], requires_grad=0, device=cuda:0) = prim::add_optional(%0, %4)
  %6 : Float(4, 2, 2, 2, strides=[8, 4, 2, 1], requires_grad=0, device=cuda:0) = aten::relu(%5) # /opt/pytorch/nvfuser/python_tests/test_torchscript.py:4118:19
  return (%6)
```
But with current TOT pytorch `2.1.0a0+git232b96b` I am seeing
```
python_tests/test_torchscript.py::TestCudaFuser::test_conv2d_bias graph(%x.1 : Tensor,
      %w.1 : Tensor,
      %bias.1 : Tensor):
  %3 : int[] = prim::Constant[value=[1, 1]]()
  %4 : int[] = prim::Constant[value=[0, 0]]()
  %5 : int = prim::Constant[value=1]()
  %o.1 : Tensor = aten::conv2d(%x.1, %w.1, %bias.1, %3, %4, %3, %5) # /opt/pytorch/nvfuser/python_tests/test_torchscript.py:4117:16
  %7 : Tensor = aten::relu(%o.1) # /opt/pytorch/nvfuser/python_tests/test_torchscript.py:4118:19
  return (%7)
```

### PR DESCRIPTION
**Pull request by jacobhinkle**
Fixes #624 (for now).

### CODE REVIEW

**Comment by jacobhinkle** [comment]
!build

**Comment by jacobhinkle** [comment]
All the torchscript tests are disabled, but there is now a dynamo test failing: `test_dtype_correctness`. That test checks that there are no warnings using `@torchdynamo.optimize("nvprims_nvfuser")` but we actually get 5:
```
{message : UserWarning("'has_cuda' is deprecated, please use 'torch.backends.cuda.is_built()'"), category : 'UserWarning', filename : '/opt/pytorch/pytorch/torch/overrides.py', lineno : 111, line : None}
{message : UserWarning("'has_cudnn' is deprecated, please use 'torch.backends.cudnn.is_available()'"), category : 'UserWarning', filename : '/opt/pytorch/pytorch/torch/overrides.py', lineno : 112, line : None}
{message : UserWarning("'has_mps' is deprecated, please use 'torch.backends.mps.is_built()'"), category : 'UserWarning', filename : '/opt/pytorch/pytorch/torch/overrides.py', lineno : 118, line : None}
{message : UserWarning("'has_mkldnn' is deprecated, please use 'torch.backends.mkldnn.is_available()'"), category : 'UserWarning', filename : '/opt/pytorch/pytorch/torch/overrides.py', lineno : 119, line : None}
{message : UserWarning('nvfuser integration in primTorch is deprecated'), category : 'UserWarning', filename : '/opt/pytorch/pytorch/torch/_prims/nvfuser_executor.py', lineno : 300, line : None}
```
cc @kevinstephano  @jjsjann123

**Comment by jacobhinkle** [comment]
!build

File: python_tests/test_torchscript.py [hunk 1]
```diff
@@ -873,7 +873,9 @@ def test_unary_ops(self):
         GRAPH_EXECUTOR != ProfilingMode.PROFILING,
         "Requires fusion optimization pass to be effective",
     )
-    def test_category_rule(self):
+    # Disabled due to change in torch.testing._internal.jit_metaprogramming_utils
+    # See https://github.com/NVIDIA/Fuser/issues/624
+    def _test_category_rule(self):
```
Comment (jjsjann123) [inline python_tests/test_torchscript.py:878]: nitpick, we can probably a proper decorator other than hacking the test name.
`@unittest.skipIf(True, "disabling TorchScript failure See https://github.com/NVIDIA/Fuser/issues/624")`

**Review by jjsjann123 — APPROVED** [review]
Thanks for going through the test failures. One small nitpick on the style.

File: python_tests/test_torchscript.py (hunk 1 above)
Comment (jacobhinkle) [reply python_tests/test_torchscript.py:878]: Oh of course. That's a good call. Will update and merge after dinner.

**Comment by jjsjann123** [comment]
!build

**Comment by jjsjann123** [comment]
!build

**Comment by jjsjann123** [comment]
!build

File: python_tests/test_torchscript.py (hunk 1 above)
Comment (jacobhinkle) [reply python_tests/test_torchscript.py:877]:
```suggestion
    def test_category_rule(self):
```

Comment (jacobhinkle) [reply python_tests/test_torchscript.py:877]: With `@skipIf`, we can name these back to `test_*`

**Review by jacobhinkle — COMMENTED** [review]
We can remove underscores from the names now that they are always skipped

File: python_tests/test_torchscript.py (hunk 1 above)
Comment (jjsjann123) [reply python_tests/test_torchscript.py:877]: Oops, somehow I missed that. I'll double check those when CI finished. Thanks for catching this one.

**Comment by jjsjann123** [comment]
Actually CI looks ok, we are only waiting on A100 nodes.

**Comment by jjsjann123** [comment]
oops. looks like we are going to miss the CI tonight :cry:

**Pull request merged**